### PR TITLE
#235 package styles.json

### LINKED ISSUE #235: Deploy style.json, glyphs, …

### LINKED ISSUE #243: Use new tilejsons
We are hosting most now on our own. Change the urls in the stlye.json

### PR DESCRIPTION
**Pull request by elKei24**
Closes #235

Closes #243

### CODE REVIEW

File: docker-compose.staging.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ services:
   reverse_proxy:
     image: "nginx:1.19.5"
     volumes:
-      - ./docker/reverse_proxy/www:/usr/share/nginx/html
+      - ./mapbox/styles:/usr/share/nginx/html
```
Comment (maxammann) [inline docker-compose.staging.yml:14]:
```suggestion
      - ./mapbox/styles:/usr/share/nginx/html
```

I would avoid the name mapbox, maybe map-styles would be better.

File: mapbox/styles/style.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
     }
   },
   "sprite": "https://tiles.staging.ehrenamtskarte.app/sprites/osm-liberty",
-  "glyphs": "https://orangemug.github.io/font-glyphs/glyphs/{fontstack}/{range}.pbf",
+  "glyphs": "https://tiles.staging.ehrenamtskarte.app/font-glyphs/glyphs/{fontstack}/{range}.pbf",
```
Comment (maxammann) [inline mapbox/styles/style.json:38]:
```suggestion
  "glyphs": "https://tiles.staging.ehrenamtskarte.app/glyphs/{fontstack}/{range}.pbf",
```

File: mapbox/styles/style.json [hunk 3]
```diff
@@ -14,9 +14,7 @@
     "natural_earth_shaded_relief": {
       "maxzoom": 6,
       "tileSize": 256,
-      "tiles": [
-        "https://klokantech.github.io/naturalearthtiles/tiles/natural_earth_2_shaded_relief.raster/{z}/{x}/{y}.png"
-      ],
+      "url": "https://maps.tuerantuer.org/natural_earth/tiles.json",
```
Comment (maxammann) [inline mapbox/styles/style.json:17]: @elKei24 I changed this url here

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
@elKei24 can you send me a deb file for this package?

File: .circleci/config.yml [hunk 4]
```diff
@@ -109,10 +111,38 @@ jobs:
...
+          path: ~/project
+      - run:
+          name: "Download glyphs"
+          command: |
+            wget -O /tmp/glyphs.zip https://github.com/orangemug/font-glyphs/archive/gh-pages.zip
+            unzip -d ./styles /tmp/glyphs.zip
+            mv ./styles/font-glyphs* ./styles/font-glyphs
+      - run:
+          name: "Build .deb"
+          command: ../scripts/pack_deb.sh -v "0.$CIRCLE_BUILD_NUM" -m ./styles -d "Mapbox styles.json with dependencies for the Ehrenamtskarte app" -n "eak-styles"
```
Comment (maxammann) [inline .circleci/config.yml:140]:
```suggestion
          command: ../scripts/pack_deb.sh -v "0.$CIRCLE_BUILD_NUM" -m ./styles -d "Mapbox styles.json with dependencies for the Ehrenamtskarte app" -n "eak-map-styles"
```

**Comment by elKei24** [comment]
> @elKei24 can you send me a deb file for this package?

Sure: https://238-310244313-gh.circle-artifacts.com/0/tmp/artifacts/debs/eak-styles_0.238-1_all.deb

**Comment by maxammann** [comment]
Nice! the styles are used now in production: https://tiles.ehrenamtskarte.app/sprites/osm-liberty.json

Is this package also in the eak meta-package?

File: docker-compose.staging.yml (hunk 1 above)
Comment (elKei24) [reply docker-compose.staging.yml:14]: I was sure someone would dislike this, I also do 😆 Hm, I find map-styles does not fit to martin. Maybe map-tiles?

File: .circleci/config.yml [hunk 5]
```diff
@@ -127,7 +157,7 @@ jobs:
     steps:
       - checkout:
           path: ~/project
-      - run: ../scripts/pack_deb.sh -v 0.1 -d "Meta package for the Ehrenamtskarte app" -n "eak" -c "eak-backend, eak-administration"
+      - run: ../scripts/pack_deb.sh -v "0.$CIRCLE_BUILD_NUM" -d "Meta package for the Ehrenamtskarte app" -n "eak" -c "eak-backend, eak-administration, eak-styles"
```
Comment (elKei24) [inline .circleci/config.yml:160]:
```suggestion
      - run: ../scripts/pack_deb.sh -v "0.$CIRCLE_BUILD_NUM" -d "Meta package for the Ehrenamtskarte app" -n "eak" -c "eak-backend, eak-administration, eak-map-styles"
```

File: mapbox/styles/style.json (hunk 2 above)
Comment (elKei24) [reply mapbox/styles/style.json:38]: @maxammann Are you sure? If we just serve the styles directory, then there is this subfolder font-glyphs on top of glyphs (as glyphs is not the only child of font-glyphs).

**Comment by elKei24** [comment]
> Is this package also in the eak meta-package?

Yes, it is 😃

**Review by maxammann — APPROVED** [review]

File: mapbox/styles/style.json (hunk 2 above)
Comment (elKei24) [reply mapbox/styles/style.json]: Done

File: docker-compose.staging.yml (hunk 1 above)
Comment (elKei24) [reply docker-compose.staging.yml]: Done

**Pull request merged**